app/src/main/java/net/programmierecke/radiodroid2/ActivityMain.java

```diff
@@ -594,10 +595,18 @@ public void onFocusChange(View v, boolean hasFocus) {
         menuItemListView.setVisible(false);
         menuItemIconsView.setVisible(false);
         menuItemAddAlarm.setVisible(false);
-        
-        MPDServersRepository mpdServersRepository = new MPDServersRepository(getApplicationContext());
-        LiveData<List<MPDServerData>> mpdServers = mpdServersRepository.getAllServers();
-        menuItemMpd.setVisible(mpdServers.getValue().size() > 0);
+
+        RadioDroidApp radioDroidApp = (RadioDroidApp) getApplication();
+        if (radioDroidApp != null) {
+            MPDClient mpdClient = radioDroidApp.getMpdClient();
+            if (mpdClient != null) {
+                MPDServersRepository repository = mpdClient.getMpdServersRepository();
+                if (repository != null) {
+                    LiveData<List<MPDServerData>> mpdServers = repository.getAllServers();
```

Comment:
I think it may be nicer if `MPDServersRepository` have `isEmpty` method. But that's just a nitpick.